### PR TITLE
After openssl is upgraded facing __GNUC__ safe_math.h not found issue

### PR DESCRIPTION
**Pull request by HemaAnmisha**
REF:https://bugzilla.tianocore.org/show_bug.cgi?id=4880

Facing the below issue after openssl is upgraded

Edk2\CryptoPkg\Library\OpensslLib\openssl\include\internal/safe_math.h(19): warning C4668: '__GNUC__' is not defined as a preprocessor macro, replacing with '0' for '#if/#elif'

CLA: trivial

Reviewed-by: Gerd Hoffmann <kraxel@redhat.com>

##### Checklist

- [x] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Review by paulidale — APPROVED** [review]
The C preprocessor is supposed to replace unknown symbols with zero.  It's not supposed to warn about doing it.

Still, this is a harmless change.

**Comment by t8m** [comment]
This could be accepted with CLA: trivial. Could you please amend the commit message to contain `CLA: trivial` on a separate line?

**Comment by paulidale** [comment]
I too would be okay with trivial for this.

**Comment by paulidale** [comment]
The style failure looks relevant unfortunately.

File: include/internal/safe_math.h
Comment (tom-cosgrove-arm) [inline include/internal/safe_math.h:21]: This line is indented by one space too many

Comment (HemaAnmisha) [reply include/internal/safe_math.h]: addressed

**Comment by t8m** [comment]
@HemaAnmisha  Could you please amend the commit message to contain `CLA: trivial` on a separate line?

**Comment by tom-cosgrove-arm** [comment]
@HemaAnmisha [All contributions to OpenSSL need to either be covered by a CLA, or to be considered trivial for copyright concerns.](https://openssl-library.org/policies/cla/) That's why you are seeing "CLA missing" and `hold: cla required`.

This change could be accepted as "CLA: trivial", which would avoid the CLA requirement. For "CLA: trivial" we need you to amend the commit message (replacing the existing commit) to have the text `CLA: trivial` on a separate line in the commit message body, separated from the rest of the commit message by an empty line. For an example, see [my comment here](https://github.com/openssl/openssl/pull/25257#pullrequestreview-2255031758).

**Comment by HemaAnmisha** [comment]
> @HemaAnmisha Could you please amend the commit message to contain `CLA: trivial` on a separate line?

done

**Review by t8m — APPROVED** [review]

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM, okay with trivial, thanks

**Review by paulidale — APPROVED** [review]
My approval stands.

**Comment by HemaAnmisha** [comment]
can this pr be merged, now?

**Comment by tom-cosgrove-arm** [comment]
> can this pr be merged, now?

The required approvals have been obtained. We just wait for 24 hours from the second approval to confirm no other objections are raised before merging.

**Pull request closed without merging**
